lib/matplotlib/legend.py

```diff
@@ -621,12 +591,27 @@ def _get_loc(self):
 
     _loc = property(_get_loc, _set_loc)
 
+    def set_loc(self, loc):
+        """
+        Set the legend location. For possible values see the `~.Axes.legend`
+        docstring.
+        """
+        self._loc_used_default = False
+        self._set_loc(loc)
+
+    def get_loc(self):
+        """
+        Get the legend location. This will be one of 'best', 'NE', 'NW',
```

Comment:
Not sure we want to promote the short compass notation as the canonical API (which it effectively is, if we use it as the return value for `get_loc`.

I'm inclined to leave out  `set_loc`/`get_loc` for now as I'm not sure what the canonical API for `get_loc` should be. And IMHO it's not a too important feature, so that not much harm is done if we're taking this slowly.

Has there been a user request for this or have you added it just because it was straight forward to implement?

Reply:
I don't know of a special user request; but users rarely ask for making private attributes public, instead they just use the private ones. As it is now, this PR does not break any user code that assigns to the private attributes, i.e. `legend._loc = 3` still works as expected. But it would break user code that would compare the location (`if legend._loc == 3: ..`). I'd say this is the risk of using private attributes, but we should provide them with a public alternative, which is `get_loc`.

Reply:
When we create new public API we should be sure how we want that, and I'm not in this case. This needs discussion with other core devs and should not slip in silently as part of a larger PR. I propose to pull this out into a seprate PR and discuss there.

Reply:
This was discussed in the call today https://paper.dropbox.com/doc/Matplotlib-2019-meeting-agenda--AYtd1qG~KP8b8TappeAo1YrhAg-aAmENlkgepgsMeDZtlsYu#:h2=12679-compass-notation:

We're not yet clear what the right return value for `get_loc` should be (return the actual user input / or normalized to long compass notation / or normalized to short compass notation).

The main part of this PR is very useful and agreed upon. It would be good get this into 3.1. Please extract `set_loc`/`get_loc` to a separate PR targeted at 3.2 so that we have a bit more time to decide how to do them right.

Reply:
Meaning you want me to introduce private `_get_loc`/`_set_loc` for `AnchoredOffsetbox` as well?

Reply:
I'm not following what you are asking here.  This PR introduces `get_loc` and `set_loc`; they are new, right?  So all we are asking is the new API be delayed so we can talk about it some more.

Reply:
OMG! I think the complete concept breaks down, since 

https://github.com/matplotlib/matplotlib/blob/da701ae6c60c92a4df709b66caf010b8f15b8c06/lib/matplotlib/offsetbox.py#L1029

was public all the time. Hence this PR would break code using the value received by `.loc`.

Reply:
Not thought this through completely: Are you saying this is a fundamental problem?

Reply:
At least it makes the current state of this PR backwards-incompatible. I need think more on that.